### ISSUE TITLE
Problem: Wayback won't start

### ISSUE THREAD

**Issue by peterVG** [webrecorder/pywb#580]
## Describe the bug
Installing `pywb` via `pip install` in a python 3.8.5 or python 3.7 virtualenv on MacOS produced this error when trying to start the Wayback server as per Getting Started instructions:
```
2020-09-24 19:44:52,499: [INFO]: Starting Gevent Server on 8080
zsh: segmentation fault  wayback
```
Got exact same result when doing a `git clone` of the repo master branch and installing via `python setup.py install`

## Steps to reproduce the bug
`pip install` or `git clone` install pywb. run `Wayback` command

## Expected behavior
Wayback server starts and is available at http://localhost:8080/

## Screenshots
<img width="466" alt="Screen Shot 2020-09-24 at 7 45 02 PM" src="https://user-images.githubusercontent.com/672121/94221156-5e165b00-fe9f-11ea-8f41-57f4f800fc69.png">

## Environment

Mac OS 10.15.6

**Comment by peterVG** [comment]
Also, no love on Docker. Also seems to have an issue with Wayback :-(

![Uploading Screen Shot 2020-09-24 at 8.10.11 PM.png…]()

**Comment by ikreymer** [comment]
I haven't seen this crash before, perhaps related to OSX 10.15, hopefully something simple.

For Docker, there's a typo, you want to run the `webrecorder/pywb` image not wayback, so should be `docker run -p 8080:8080 -v /pywb-data:/webarchive webreccorder/pywb`

**Comment by peterVG** [comment]
Okay, thanks. Will retry Docker and let you know 

In the meanwhile, I spun up an Ubuntu 20.04 server. Installed from Git. Using Python 3.8.2. Switched to `v-2.4.2-beta` tag to be sure. But once again ran into a `segmentation fault` error:

<img width="1213" alt="Screen Shot 2020-09-24 at 8 42 03 PM" src="https://user-images.githubusercontent.com/672121/94224111-6920b980-fea6-11ea-939c-7dac5c74e392.png">

**Comment by peterVG** [comment]
So the Docker server started but then it wasn't returning results for the sites in the WARC file I thought I initialized it with. 

<img width="1060" alt="Screen Shot 2020-09-24 at 9 04 18 PM" src="https://user-images.githubusercontent.com/672121/94225551-21039600-feaa-11ea-9286-33c8b478807b.png">

When I checked in the project directory the WARC file was gone. So I tried to re-initialize with a new (Common Crawl) WARC but now I get the error:
`Error: Collection default does not exist. To create a new collection, run
/usr/local/bin/wb-manager init default` 

I seemed to have run that command successfully but it repeats the same error:

<img width="1232" alt="Screen Shot 2020-09-24 at 9 04 02 PM" src="https://user-images.githubusercontent.com/672121/94225529-16e19780-feaa-11ea-88b7-cf37f6eb0e23.png">

**Comment by peterVG** [comment]
I deleted the project directory and did a fresh `git clone`. I switched to the stable `v-2.4.2-beta` tag. This time I made sure my WARC file was in the specified location. I bind mounted the `/Users/peter/Development/pywb/pywb-data` directory using the Docker Desktop > Preferences > Resources > File Sharing menu. Am I missing a step with the `webarchive` directory? I am not totally clear what is intended by the `/pywb-data:/webarchive` portion of the Docker command:

<img width="1140" alt="Screen Shot 2020-09-24 at 9 34 19 PM" src="https://user-images.githubusercontent.com/672121/94228031-d2a5c580-feb0-11ea-964c-9b38bb57637b.png">

I realized that this was just a warning about a missing default, not a fatal error:
`Error: Collection default does not exist. To create a new collection, run
/usr/local/bin/wb-manager init default`

I successfully started the server via the Docker container:

`(venv) peter@Peters-MacBook-Air pywb % docker run -p 80:8080 -v /Users/peter/Development/pywb/pywb-data/:/webarchive  webrecorder/pywb  `

I successfully ran `warcio` on the WARC file to make sure the URLs I was searching for are in the WARC. However, I am not getting any hits in that WARC:

<img width="1126" alt="Screen Shot 2020-09-24 at 9 45 55 PM" src="https://user-images.githubusercontent.com/672121/94227508-7ee6ac80-feaf-11ea-998a-840377bbc500.png">

So I switched the test WARC file to use `/sample_data/warc/example.warc.gz`
<img width="1036" alt="Screen Shot 2020-09-24 at 10 03 35 PM" src="https://user-images.githubusercontent.com/672121/94228453-dab23500-feb1-11ea-9f40-6415674e593f.png">

warcio tells me that it contains the url `http://example.com?example=1`. But again no dice:

<img width="1069" alt="Screen Shot 2020-09-24 at 10 02 19 PM" src="https://user-images.githubusercontent.com/672121/94228480-f0bff580-feb1-11ea-990c-640e7e0fef92.png">

**Comment by peterVG** [comment]
I switched to running a fresh `git clone` install on my MacOS using a Python 2.7 environment. No `segmentation fault` error this time but a new one:

![Uploading Screen Shot 2020-09-24 at 11.04.06 PM.png…]()

**Comment by peterVG** [comment]
I realize I'm reporting on a number of different environments/deployments here. I really just wanted to test that I could read a WARC on my Mac laptop before doing an Ubuntu server install (the WebRecorder Electron Player also crashed on the Common Crawl WARC BTW). 

A production Ubuntu/uwsgiNginx install is really my end goal so if do you have time to investigate further my main concern is getting past any Ubuntu roadblocks and getting pywb running on it.

**Comment by jamadden** [comment]
The crashes appear to be because of a version mismatch between gevent and greenlet. On Python 3.7+, using greenlet 0.4.17 requires gevent 20.9.0 (and 20.9.0 requires 0.4.17). 

See https://github.com/gevent/gevent/issues/1678 and https://github.com/python-greenlet/greenlet/issues/178

**Comment by peterVG** [comment]
Hmm. Thanks for looking into this @jamadden. However, when I change the gevent version to 20.9.0. I get an error running the `wayback` command which notes that `The 'gevent==1.4.0' distribution was not found and is required by pywb`

<img width="1227" alt="Screen Shot 2020-09-25 at 7 50 56 AM" src="https://user-images.githubusercontent.com/672121/94281878-065e0b00-ff04-11ea-9595-bca6b7f6674f.png">

**Comment by jamadden** [comment]
I don't know anything about `pywb`, but I think you'll also need to change (at least relax) the requirement and reinstall pywb instead of just changing the version out from under it.

Or you could install an alder version of greenlet.

**Comment by peterVG** [comment]
@jamadden okay, yeah, I just changed the Python version to 3.5 on my Ubuntu install using the deadsnakes repo and then tried again but same `segmentation fault` error. However, I didn't have a fresh `pywb` install so I'll try that now.

<img width="1395" alt="Screen Shot 2020-09-25 at 8 10 12 AM" src="https://user-images.githubusercontent.com/672121/94284131-cf3d2900-ff06-11ea-8310-2c0ff6c55a07.png">

**Comment by peterVG** [comment]
Fresh `git clone` on Ubuntu 20.04. Switched to a Python 3.5 virtualenv. Running `python setup.py install` gets me the error: 

`  File "/tmp/easy_install-ldgzarg1/MarkupSafe-2.0.0a1/src/markupsafe/__init__.py", line 83
    return f"{self.__class__.__name__}({super().__repr__()})"
                                                            ^
SyntaxError: invalid syntax
(venv) root@ubuntu-s-1vcpu-1g`

**Comment by peterVG** [comment]
SUCCES AT LAST!!

A new clean project directory on Ubuntu 20.04. Switched to a Python 3.5 virtualenv. Ran `pip install pywb`. 

Then:

```
wb-manager init my-web-archive
wb-manager add my-web-archive common_crawl/CC-MAIN-20200803083123-20200803113123-00000.warc.gz 
wayback
```

I can now query and render results for that WARC using pywb on port 8080!

**Comment by ikreymer** [comment]
Sorry you were having so much issues. Yes, this seems to be a problem with gevent, just started seeing this recently.
I'll see about doing a quick update to later version of gevent. Glad you got it working in the end!

**Comment by codl** [comment]
Hi, I had the same (or a similar) issue:

```
$ wayback --port 60720                                                                                                                                                      
<frozen importlib._bootstrap>:241: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 152 from C header, got 160 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 152 from C header, got 160 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 152 from C header, got 160 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 152 from C header, got 160 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 152 from C header, got 160 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 152 from C header, got 160 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 152 from C header, got 160 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 152 from C header, got 160 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 152 from C header, got 160 from PyObject
<frozen importlib._bootstrap>:241: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 152 from C header, got 160 from PyObject
fish: Job 1, 'wayback --port 60720' terminated by signal SIGSEGV (Address boundary error)
```

Manually installing the latest version of gevent with `pip install gevent==21.12.0` seemed to fix it.

**Comment by edsu** [comment]
I'm just confirming that `gevent 21.12.0` seemed to fix things for me on OSX 12.3.1.
